### PR TITLE
Carousel is Implemented in the Featured Channels Sections in dashboard

### PR DESCRIPTION
**Pull request by abhishek-01k**
### Ticket Number

#1642 

### Description

In the Featured Channels Section in dashboard, Implemented the Carousel instead of scrollbar. Now 3 slides will be scrolled when next or prev is clicked. 

Used Embla Carousel for this process, https://www.embla-carousel.com/get-started/.

Mobile viewport is also done along with laptop and tabled viewport.

- **Problem/Feature**:

### Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

- [ ] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [ ] Quick PR label added
- [ ] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [ ] DO NOT Merge PR label added

### Frontend Guidelines

- [ ] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [ ] No errors in the build terminal
- [ ] Engineer has tested the changes on their local environment
- [ ] Engineer has tested the changes on deploy preview

### Screenshots/Video with Explanation

Screenshot 2024-07-08 at 9 16 37 AM

### Review & Approvals

- [ ] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### CODE REVIEW

File: src/modules/dashboard/components/FeaturedChannels.tsx [hunk 1]
```diff
@@ -47,66 +30,18 @@ const FeaturedChannels: FC<FeaturedChannelsProps> = () => {
...
           featuredChannelsList={mobileFeaturedChannelsList}
         />
+
       ) : (
         <FeaturedChannelsList
-          listRef={listRef}
           featuredChannelsList={featureChannelsForCurrrentEnv}
         />
       )}
+
```
Comment (rohitmalhotra1420) [inline src/modules/dashboard/components/FeaturedChannels.tsx:44]: I think the FeaturedChannelsList.tsx and FeaturedChannelsMobileView.tsx can be combined into one component called  FeaturedChannelsList.tsx for mobile small, mobile medium, mobile large, laptop, laptop large and desktop screen sizes.

We might need a separate FeaturedChannelsTabletView.tsx component. Please see what approach works best.

Comment (rohitmalhotra1420) [inline src/modules/dashboard/components/FeaturedChannels.tsx:44]: **UI Broken for tablet view**

Screenshot 2024-07-08 at 5 13 41 PM



**For mobile small the spacing is off on the right side.**

Screenshot 2024-07-08 at 5 13 51 PM



**For mobile medium the spacing is off on the right side.**

Screenshot 2024-07-08 at 5 13 57 PM



**For mobile large the spacing is off on the right side.**

Screenshot 2024-07-08 at 5 14 02 PM



**For laptop the spacing is off on the right side and item width doesn't matches the design**

Screenshot 2024-07-08 at 5 14 11 PM



**For desktop the spacing is off on the right side and the width doesn't matches the design**

Screenshot 2024-07-08 at 5 14 27 PM

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]
Needs fixes. Please take a pull before implementing the fixes.

File: src/modules/dashboard/components/FeaturedChannels.tsx (hunk 1 above)
Comment (abhishek-01k) [reply src/modules/dashboard/components/FeaturedChannels.tsx:44]: FeaturedChannels is divided into two components. FeaturedChannels and FeaturedChannelsMobileView. 

Laptop and tablet will be rendered into FeaturedChannels and mobile will be rendered with FeaturedChannelsMobile. 

In the tablet mode, 2 slides will be shown and on clicking next 2 slides will be shown. In the laptop mode, 3 slides will be shown and on clicking next 3 slides will be shown.

In the mobile view, 3 channels are rendered on a single viewport vertically and upon clicking next, next slide is shown.

**Comment by abhishek-01k** [comment]
Screenshot 2024-07-10 at 10 48 34 AM
Screenshot 2024-07-10 at 10 48 42 AM
Screenshot 2024-07-10 at 10 48 49 AM
Screenshot 2024-07-10 at 10 49 09 AM
Screenshot 2024-07-10 at 11 28 35 AM

**Review by rohitmalhotra1420 — APPROVED** [review]

**Pull request merged**